proposed/vector-instruction-set-default.md

```diff
@@ -0,0 +1,100 @@
+
+## Differ behavior by OS
+
+It is reasonable to vary SIMD defaults, per operating system. There are many aspects of the product that differ by operating system or hardware platform. In fact, the JIT dynamically targeting different SIMD instructions per machine is such an example. For this case, the key consideration is the ready-to-run native code produced for a given operating system + CPU consideration. We could make one decision for Linux x64, another for Windows x64, and another for macOS Arm64. We already have to target different SIMD instructions for x64 vs Arm64, so targeting different SIMD instructions across OSes isn't much of a leap.
+
+It is useful to look at hardware dates again. Continuing with the Intel focus, we can see the following delivery of SIMD instructions in chips:
+
+* 2006: SSE4 was introduced with Core 2 x64 chips at the initial [Intel Core architecture](https://en.wikipedia.org/wiki/Intel_Core_(microarchitecture)) launch.
+* 2008: SSE4.2 was introduced with [Nehalem](https://en.wikipedia.org/wiki/Nehalem_(microarchitecture)) chips.
+* 2011: AVX was introduced with [Sandy Bridge](https://en.wikipedia.org/wiki/Sandy_Bridge) chips.
+* 2013: AVX2 was introduced with [Haswell](https://en.wikipedia.org/wiki/Haswell_(microarchitecture)) chips.
+
+Let's take a look at each operating system, assuming the Intel ISA.
+ 
+* On Windows, we need to be conservative. There are a lot of .NET users on Windows, and the majority of .NET desktop apps target Windows. Windows 10 requires AVX, but Win7 requires only SSE2. It is reasonable to expect that we'll continue supporting Windows 7 with .NET 6.0 (and probably not after that).
+* On [macOS](https://en.wikipedia.org/wiki/MacOS_version_history#Version_10.13:_%22High_Sierra%22), it appears that [Mac machines have had AVX2 since late 2013](https://en.wikipedia.org/wiki/List_of_Macintosh_models_grouped_by_CPU_type#Haswell), when they adopted [Haswell chips](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX2). In terms of macOS, [macOS Big Sur (11.0)](https://support.apple.com/kb/sp833?locale=en_US) appears to be the first version to require Haswell. .NET 5.0, for example, supports [macOS High Sierra (10.13)](https://support.apple.com/kb/SP765?locale=en_US) and later. macOS 10.13 supports hardware significantly before Haswell. For .NET 6.0, we'll likely continue to choose a "10." macOS version as the minimum, possibly [macOS Catalina (10.15)](https://support.apple.com/kb/SP803?locale=en_US), aligning with our past practice of slowly (not aggressively) moving forward OS requirements. Zooming out, macOS x64 is now a legacy platform. We shouldn't rock the boat with a change that could cause a performance regression at this late stage of the macOS x64 lifecycle.
```

Comment:
Good call on macOS. Macs are expensive and people hold onto them until they die. We have mac machines in our own dotnet CI build pools that don't have AVX2.

I learned the hard way:

https://github.com/MichalStrehovsky/runtimelab/blob/9752e8627cf08c60efbd6f21faecd7d098ad5581/src/tests/nativeaot/SmokeTests/HardwareIntrinsics/x64Vex.csproj#L24-L32

Reply:
> people hold onto them until they die.

Or in my case, I'm still holding onto a Macbook Pro from ~2011/2012 because it has AVX but no AVX2 support. This proved immensely useful at catching/repro'ing bugs when first introducing the x86 hardware intrinsics.